test/e2e_cri/container.go

```diff
@@ -0,0 +1,147 @@
+	podConfig := &runtimeapi.PodSandboxConfig{
+		Metadata: buildPodSandboxMetadata(&podName),
+	}
+	podID, err := c.RunPodSandbox(podConfig)
+	framework.ExpectNoError(err, "Failed to create PodSandbox: %v", err)
+	framework.Logf("Created PodSandbox %s\n", podID)
+	return podID, podConfig
+}
+
+func testCreateContainer(c internalapi.RuntimeService, podID string, podConfig *runtimeapi.PodSandboxConfig) string {
+	By("create container")
+	containerName := "container-for-create-test-" + string(uuid.NewUUID())
+	containerConfig := &runtimeapi.ContainerConfig{
+		Metadata: buildContainerMetadata(&containerName),
+		Image:    &runtimeapi.ImageSpec{Image: &imageName},
+		Command:  []string{"sh", "-c", "top"},
+	}
+	containerID, err := c.CreateContainer(podID, containerConfig, podConfig)
+	framework.ExpectNoError(err, "Failed to create container: %v", err)
+	framework.Logf("Created container %s\n", containerID)
+
+	By("get container status")
+	status, err := c.ContainerStatus(containerID)
+	framework.ExpectNoError(err, "Failed to get container %s status: %v", containerID, err)
+	Expect(*status.State).To(Equal(runtimeapi.ContainerState_CONTAINER_CREATED), "Container state should be created")
+
+	return containerID
+}
+
+func testStartContainer(c internalapi.RuntimeService, containerID string) {
+	By("start container")
+	err := c.StartContainer(containerID)
+	framework.ExpectNoError(err, "Failed to start container: %v", err)
+	framework.Logf("Start container %s\n", containerID)
+
+	By("get container status")
+	status, err := c.ContainerStatus(containerID)
+	framework.ExpectNoError(err, "Failed to get container %s status: %v", containerID, err)
+	Expect(*status.State).To(Equal(runtimeapi.ContainerState_CONTAINER_RUNNING), "Container state should be started")
+}
+
+var _ = framework.KubeDescribe("Test CRI container", func() {
+	f := framework.NewDefaultCRIFramework("CRI-container-test")
+
+	var c internalapi.RuntimeService
+
+	BeforeEach(func() {
+		c = f.CRIClient.CRIRuntimeClient
+	})
+	Context("basic container operatings", func() {
+		var podID string
+		var podConfig *runtimeapi.PodSandboxConfig
+
+		BeforeEach(func() {
+			podID, podConfig = createPodSandbox(c)
+		})
+
+		AfterEach(func() {
+			By("stop PodSandbox")
+			c.StopPodSandbox(podID)
+			By("delete PodSandbox")
+			c.RemovePodSandbox(podID)
+		})
+
+		It("test create container", func() {
+			testCreateContainer(c, podID, podConfig)
+		})
+
+		It("test start container", func() {
+			containerID := testCreateContainer(c, podID, podConfig)
+			testStartContainer(c, containerID)
+		})
+
+		It("test stop container", func() {
+			containerID := testCreateContainer(c, podID, podConfig)
+			testStartContainer(c, containerID)
+
+			By("stop container")
+			err := c.StopContainer(containerID, 60)
+			framework.ExpectNoError(err, "Failed to stop container: %v", err)
+			framework.Logf("Stop container %s\n", containerID)
+
+			By("get container status")
```

Comment:
@pskrzyns Any thing else would you like to refact, maybe I could do it in my pr directly.